tests/test_check_parallel.py

```diff
@@ -0,0 +1,141 @@
+        self.check_strings(cmd, output + err, expected_present, expected_absent)
+        # Put the original index.db back.
+        os.remove('zstash/index.db')
+        shutil.copy('zstash/index_old.db', 'zstash/index.db')
+        os.chdir(TOP_LEVEL)
+
+        print('Verifying the data from database with the actual files')
+        # Checksums from HPSS
+        cmd = ['sqlite3', 'zstash_test/zstash/index.db', 'SELECT md5, name FROM files;']
+        output_hpss, err_hpss = run_cmd(cmd)
+        hpss_dict = {}
+
+        for l in output_hpss.split('\n'):
+            l = l.split('|')
+            if len(l) >= 2:
+                f_name = l[1]
+                f_hash = l[0]
+                hpss_dict[f_name] = f_hash
+
+        # Checksums from local files
+        cmd = '''find zstash_test_backup -regex .*\.txt.* -exec md5sum {} + '''
+        output_local, err_local = run_cmd(cmd)
+        local_dict = {}
+
+        for l in output_local.split('\n'):
+            l = l.split('  ')
+            if len(l) >= 2:
+                f_name = l[1].split('/')  # remove the 'zstash_test_backup'
+                f_name = '/'.join(f_name[1:])
+                f_hash = l[0]
+                local_dict[f_name] = f_hash
+        print('filename|HPSS hash|local file hash')
+        for k in local_dict:
+            print('{}|{}|{}'.format(k, hpss_dict[k], local_dict[k]))
+
+    def testCheckParallelVerbose(self):
+        self.helperCheckParallel('testCheckParallelVerbose', 'none', verbose=True)
+
+    @unittest.skipIf(os.system('which hsi') != 0, 'This system does not have hsi')
+    def testCheckParallelVerboseHPSS(self):
+        self.helperCheckParallel('testCheckParallelVerboseHPSS', 'zstash_test', verbose=True)
+
+    def testCheckParallel(self):
+        self.helperCheckParallel('testCheckParallel', 'none')
+
+    @unittest.skipIf(os.system('which hsi') != 0, 'This system does not have hsi')
+    def testCheckParallelHPSS(self):
+        self.helperCheckParallel('testCheckParallelHPSS', 'zstash_test')
+        
+    def testCheckParallelVerboseMismatch(self):
+        self.helperCheckParallelVerboseMismatch('testCheckParallelVerboseMismatch', 'none')
+
+    @unittest.skipIf(os.system('which hsi') != 0, 'This system does not have hsi')
+    def testCheckParallelVerboseMismatchHPSS(self):
+        self.helperCheckParallelVerboseMismatch('testCheckParallelVerboseMismatchHPSS', 'zstash_test')
+
+    def testKeepTarsWithPreviouslySetHPSS(self):
+        """
+        Test `zstash check` in parallel when hpss is set in `zstash create`.
+        """
+        print_in_box('testKeepTarsWithPreviouslySetHPSS')
+        if os.path.exists('zstash'):
+            shutil.rmtree('zstash')
+        os.mkdir('zstash')
+        zstash_dir = ZSTASH_DIR
+        self.hpss_path = 'none'
+        # Run `zstash create`
+        run_cmd('{}zstash create --hpss={}  --maxsize 128 .'.format(zstash_dir, self.hpss_path))
```

Comment:
@golaz This test seems to require the test directory to be `.`. If I try to change it (along with the necessary file path changes in assertions), the directory `zstash` will contain `index.db` but NOT `000000.tar`. I can't figure out why that would be the case.